edisgo/tools/config.py

```diff
@@ -129,6 +138,102 @@ def __init__(self, **kwargs):
                 filename=kwargs.get("json_filename", None),
                 from_zip_archive=kwargs.get("from_zip_archive", False),
             )
+        self._config_dict = {}
+
+    @property
+    def db_table_mapping(self):
+        if not self._config_dict.get("db_table_mapping"):
+            self._set_db_mappings()
+        return self._config_dict.get("db_table_mapping", {})
+
+    @db_table_mapping.setter
+    def db_table_mapping(self, value):
+        self._config_dict["db_table_mapping"] = value
+
+    @property
+    def db_schema_mapping(self):
+        if not self._config_dict.get("db_schema_mapping"):
+            self._set_db_mappings()
+        return self._config_dict.get("db_schema_mapping", {})
+
+    @db_schema_mapping.setter
+    def db_schema_mapping(self, value):
+        self._config_dict["db_schema_mapping"] = value
+
+    def _set_db_mappings(self) -> None:
+        """
+        Sets the database table and schema mappings by retrieving alias dictionaries.
+        """
+        name_mapping, schema_mapping = self.get_database_alias_dictionaries()
+        self.db_table_mapping = name_mapping
+        self.db_schema_mapping = schema_mapping
+
+    def get_database_alias_dictionaries(self) -> tuple[dict[str, str], dict[str, str]]:
+        """
+        Retrieves the database alias dictionaries for table and schema mappings.
+
+        Returns
+        -------
+        tuple
+            A tuple containing two dictionaries:
+            - name_mapping: A dictionary mapping source table names to target table
+                names.
+            - schema_mapping: A dictionary mapping source schema names to target schema
+                names.
+        """
+        OEP_CONNECTION = "postgresql+oedialect://:@{platform}"
+        platform = "toep.iks.cs.ovgu.de"
+        conn_str = OEP_CONNECTION.format(platform=platform)
+        engine = sa.create_engine(conn_str)
+        schema_name = "model_draft"  # Replace with the actual schema name if needed
+        module_name = f"saio.{schema_name}"
+        register_schema(schema_name, engine)
+        dictionary_table_name = "edut_00"
+        dictionary_table = importlib.import_module(module_name).__getattr__(
+            dictionary_table_name
+        )
+        with session_scope_egon_data(engine) as session:
+            query = session.query(dictionary_table)
+            dictionary_entries = query.all()
+            name_mapping = {
+                entry.source_name: entry.target_name for entry in dictionary_entries
+            }
+            schema_mapping = {
+                entry.source_schema: getattr(entry, "target_schema", "model_draft")
```

Comment:
Should the string `"model_draft"` be replaced by the variable `schema_name`?

Reply:
The "target_schema" column does not exist at the moment. Until it is added, we will use "model_draft" as the default schema because all tables are currently in that schema.